scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -4534,7 +4534,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val paramss = paramClauses(ownerIsType = true, owner == OwnedByCaseClass)
       Ctor.Primary(mods, name, paramss)
     } else if (owner.isTrait) {
-      Ctor.Primary(Nil, atPos(in.tokenPos, in.tokenPos)(Name.Anonymous()), Nil)
+      Ctor.Primary(Nil, atPos(in.tokenPos, in.tokenPos - 1)(Name.Anonymous()), Nil)
```

Comment:
@dos65 quick question: below, on lines 4546 and 4557 (and maybe elsewhere), the same construct was used. should they be changed there as well? if Name.Anonymous should never have any tokens, should we define a shortcut:
```
  def atNameAnonymous(in: ...): ... = atPos(x, x - 1)(...)
```
(and what does `autoPos` on 4533 do?)

Reply:
oops)

Good catch, thx! I will fix these cases too

> and what does autoPos on 4533 do?

:thinking:

Reply:
Should do a `-1` range correctly in that case I think.